lib/using_restricted_array.rb

```diff
@@ -1,61 +1,115 @@
-require_relative 'restricted_array.rb'
+require_relative "restricted_array.rb"
 # RestrictedArray can be created using a specified size, or a random size in
 # the range of 1-20 will be chosen for you.
 # All values are integers in the range of 1-221.
 # RestrictedArray cannot be resized.
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear or O(n), where n is the number of items in the array
+#                                          or the length of the array
+# Space complexity: O(1) because there's only one integer created no matter
+#                                           how much the array's length changes
 def length(array)
-  raise NotImplementedError
+  i = 0
+  while array[i] != nil
+    i += 1
+  end
+  return i
 end
 
 # Prints each integer values in the array
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear or O(n), where n is the number of items in the array
+#                                          or the length of the array
+# Space complexity: O(1) because there's only one integer created no matter
+#                                           how much the array's length changes
 def print_array(array)
-  raise NotImplementedError
+  len = length(array)
+  i = 0
+  while i < len
+    print i
```

Comment:
Was this method supposed to print the contents of the array? If so, You'd want something like 
`print "#{array[i]} "`